proxy/proxy/salt-broker/salt-broker

```diff
@@ -348,6 +349,29 @@ if __name__ == "__main__":
             saltbroker_opts.update({"master": rhn_parent})
 
         saltbroker_opts.update(config)
+
+        formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+        # log to file or to standard output and error depending on the configuration
+        if saltbroker_opts.get('log_to_file'):
+            fileloghandler = logging.handlers.RotatingFileHandler(
+                SALT_BROKER_LOGFILE, maxBytes=200000, backupCount=5)
+            fileloghandler.setFormatter(formatter)
+            log.addHandler(fileloghandler)
+        else:
+            # prepare two log handlers, 1 for stdout and 1 for stderr
+            stdout_handler = logging.StreamHandler(sys.stdout)
+            stderr_handler = logging.StreamHandler(sys.stderr)
+            # stdout handler filter out everything below the ERROR level
+            stdout_handler.addFilter(LogFilter(logging.ERROR))
+            # stderror handler looks only for everything below the ERROR level included
+            stderr_handler.setLevel(logging.ERROR)
```

Comment:
I'm not sure I understand this. I see the following behavior after copying the logging setup to a file to play with:
```
✔ uyuni/uyuni-main % python3 log_test.py 1>log_stdout.txt 2>log_stderr.txt  
✔ uyuni/uyuni-main % cat log_stdout.txt 
2021-11-18 15:32:39,161 DEBUG Debug message                                                                         
2021-11-18 15:32:39,161 INFO Info message                                                                           
2021-11-18 15:32:39,161 WARNING Warning message
✔ uyuni/uyuni-main % cat log_stderr.txt 
2021-11-18 15:32:39,161 ERROR Error message
2021-11-18 15:32:39,161 CRITICAL Critical message 
```

Is that what you want to do? Send  `DEBUG - WARNING` to `stdout` and `ERROR + CRITICAL` to `stderr`? I think what confused me was the comment, saying "filter out everything below the ERROR level". I understood "filter out" to mean "drop messages". 

BTW, it is also possible to pass a callable (e.g. a function) that takes a single argument to `addFilter`, without creating a class for it. Just in case you didn't know, it could like like
```python
stdout_handler.addFilter(lambda record: record.levelno < logging.ERROR)
```

Reply:
Based on the [logging levels values](https://docs.python.org/3/library/logging.html#logging-levels) I think I mislead you :) so I will rephrase it with `filter out everything above the ERROR level`. The desired behavior is exactly what you described:
- stdout will receive DEBUG, INFO, WARNING
- stderr will receive ERROR, CRITICAL

And I am adopting your suggestion about the lambda over a class 

Thanks for pointing it out, fixing now :+1: